### DISCUSSION TITLE
supabase sql query times out

### DISCUSSION THREAD

**Discussion by adi44** [supabase/supabase#29753] [Questions]
Hi I am trying to run small query however everytime I try it timesout
SELECT c.text, c.parent_cast_fid, r.* 
FROM public.casts c
JOIN public.reactions r 
  ON c.parent_cast_fid = r.target_cast_fid
WHERE c.fid = 475488 
  AND c.text LIKE '%$DEGEN%'
LIMIT 100;  Hi i am trying to r

**Comment by GaryAustin1** [comment]
Probably need to provide more info.  What size tables?  Are you doing impersonation and using RLS? 
You might try explain analyze.

Like is not the fastest operation so depends on your table size.
Indexes on fid or text?

**Reply by adi44** [reply]
![image](https://github.com/user-attachments/assets/f821602f-f30c-4c43-b984-b016f3abd763)

this is info on table size

**Reply by adi44** [reply]
I am directly making call from supabase console using SQL editor

**Reply by GaryAustin1** [reply]
You can still make calls with impersonation from the SQL editor that would  need to meet RLS.

You may need to work on indexing as those are large tables and as I said LIKE is slow.

https://supabase.com/docs/guides/database/postgres/timeouts
https://supabase.com/docs/guides/database/query-optimization

https://www.cybertec-postgresql.com/en/postgresql-more-performance-for-like-and-ilike-statements/
